server/models/User.coffee

```diff
@@ -445,14 +464,18 @@ UserSchema.pre('save', (next) ->
   next()
 )
 
-UserSchema.post 'save', (doc) ->
-  doc.newsSubsChanged = not _.isEqual(_.pick(doc.get('emails'), mail.NEWS_GROUPS), _.pick(doc.startingEmails, mail.NEWS_GROUPS))
-  UserSchema.statics.updateServiceSettings(doc)
-
+UserSchema.post 'save', co.wrap ->
+  try
+    yield @updateServiceSettings()
+  catch e
+    console.error 'User Post Save Error:', e.stack
 
-UserSchema.post 'init', (doc) ->
-  doc.wasTeacher = doc.isTeacher()
-  doc.startingEmails = _.cloneDeep(doc.get('emails'))
+  
+UserSchema.post 'init', ->
+  @set('anonymous', false) if @get('email')
+  @originalEmail = @get('emailLower')
+  @wasTeacher = @isTeacher()
+  @startingEmails = _.cloneDeep(@get('emails'))
```

Comment:
Yeah, it's a bit weird... cause a couple are properties, and one is the result of a method. Might make it into a mongoose virtual property. Anyway, gonna punt this in favor of shipping sooner, but I think this'd be good to do eventually.